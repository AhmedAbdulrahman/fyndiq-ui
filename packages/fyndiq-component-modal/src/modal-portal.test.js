import React from 'react'
import { shallow } from 'enzyme'
import ModalPortal from './modal-portal'

describe('ModalPortal Component', () => {
  it('should return a Portal', () => {
    const Component = shallow(<ModalPortal />)
    expect(Component).toMatchSnapshot()
    // NOTE: right now Enzyme doesn't support Portal rendering
    // so the snapshot is `undefined`
  })

  it('should lock the body when opening the modal', () => {
    const Component = shallow(<ModalPortal />)
    expect(document.body.classList.contains('bodyFixed')).toBe(false)
    Component.setProps({ open: true })
    expect(document.body.classList.contains('bodyFixed')).toBe(true)
  })

  it('should create a portal div on mount', () => {
    shallow(<ModalPortal portalId="test-id" />)
    expect(document.getElementById('test-id')).not.toBe(null)
  })
})
