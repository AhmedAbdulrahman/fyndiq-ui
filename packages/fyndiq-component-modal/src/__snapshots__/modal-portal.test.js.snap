// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalPortal Component should return a Portal 1`] = `undefined`;
